Bynder/Sample/UploadSample.cs

```diff
@@ -41,9 +43,83 @@ private async Task RunUploadSampleAsync()
                 return;
             }
 
-            await assetService.UploadFileAsync(new UploadQuery { Filepath = uploadPath, BrandId = brands.First().Id });
+            Console.WriteLine("Name of the media item after upload: ");
+            var name = Console.ReadLine();
+            if (string.IsNullOrEmpty(name)) 
+            {
+                name = null; 
+            }
+
+            Console.WriteLine("Override original filename (leave empty to use the actual filename): ");
+            var filename = Console.ReadLine();
+
+            Console.WriteLine("Description (leave empty to use default): ");
+            var description = Console.ReadLine();
+
+            var customParameters = GetCustomParameters();
+
+            Console.WriteLine("Do you want to pass a file stream to the SDK?: (y/n)");
+            var passAsStream = Console.ReadLine().ToLower().StartsWith("y");
+
+
+            var query = new UploadQuery
+            {
+                Filepath = uploadPath,
+                BrandId = brands.First().Id,
+                Name = name,
+                CustomParameters = customParameters
+            };
+            if (!string.IsNullOrEmpty(filename))
+            {
+                query.OriginalFileName = filename;
+            }
+            if (!string.IsNullOrEmpty(description))
+            {
+                query.Description = description;
+            }
+
+           
+            query.MetapropertyOptions = new Dictionary<string,IList<string>>()
```

Comment:
@quirijnslings Was running the sample code and realized the MetapropertyOptions needed to be a relevant option, assuming the "a" key is a placeholder to be replaced with corresponding option?

Reply:
Sorry, I don't understand what you're asking.

Reply:
When running with my portal configurations, using

```
            query.MetapropertyOptions = new Dictionary<string,IList<string>>()
            {
                { "a", [ "b" ]}
            };
```

gave me a 400 during upload, but this is because I needed to replace the "a" key with an option available in my portal such as 

```
            query.MetapropertyOptions = new Dictionary<string,IList<string>>()
            {
                { "922C4766-C8ED-49E9-A5FCD5D0932E459B", [ "e0e31f07-1b37-49c5-b18d-f214708414e4" ]}
            };
 ```
 
 Wanted to clarify the "a" key serves as a placeholder for a user to provide their corresponding option id right?

Reply:
It can be used with IDs or with names. This part of the FindMediaSample illustrates this:

` 
// ...
var selectedOption = sortedOptions.Skip(mpNr - 1).FirstOrDefault();
 searchString = selectedOption.Name;

 Console.WriteLine($"Searching via the meta property named {selectedMetaProperty.Name} and option named {searchString}");
 var assets = await _bynderClient.GetAssetService().GetMediaListAsync(new MediaQuery()
 {
     MetaProperties = new Dictionary<string, IList<string>>
     {
         {
             selectedMetaProperty.Name, [ searchString ]
         }
     }
 });
// ..`

By the way, even if I search for "a": [ "b" ], I get a 200 back, never 400. Are you sure there wasn't something else going wrong?

Reply:
With the usage here I believe it is trying to assign the metaproperties on upload. In my test if I comment out query.MetapropertyOptions or change the key from "a" to my property ID I can upload, asides from that was receiving 400. But think it's fine since it's sample code.

Reply:
I misread your comment, I thought you were talking about MediaQuery.MetaProperties instead of UploadQuery.MetapropertyOptions.

What I wrote about MediaQuery is true. The similar MetapropertyOptions property of UploadQuery works differently. The difference is caused by the different behavior of the corresponding endpoints in the API.

In UploadQuery.MetapropertyOptions, the key must be the ID of a meta property (names are not accepted). But the value is a list of EITHER  option IDs OR option names.

I updated UploadSamples to illustrate this.

I also updated FindAssetSamples, it now allows you to try various key/value pairs to test how to search by meta properties / options.